src/Estimators/StructureFactorEstimator.cpp

```diff
@@ -17,6 +17,15 @@
 namespace qmcplusplus
 {
 
+StructureFactorEstimator::StructureFactorEstimator(const StructureFactorInput& sfi,
+                                                   const PSPool& pset_pool,
+                                                   DataLocality data_locality)
+    : StructureFactorEstimator(sfi,
+                               getParticleSet(pset_pool, sfi.get_source()),
+                               getParticleSet(pset_pool, sfi.get_target()),
+                               data_locality)
+{}
```

Comment:
Likely we can get remove this constructor and just keep the other one once we clean up `getParticleSet`.

Reply:
> I actually think this constructor is more fundamental that the public one.
This is the public one currently. Could you clarify?

I feel the current public one is less fundamental than the private one that was previously public and now private.